test/WebJobs.Script.Tests/NodeEndToEndTests.cs

```diff
@@ -703,13 +703,26 @@ public void ExcludedFunction_NotAddedToHost()
                     { "input", input.ToString() }
                 });
 
-            await TestHelpers.Await(() =>
+            await Task.WhenAny(t, Task.Delay(5000));
+            if (!t.IsCompleted) throw t.Exception;
+        }
+
+        [Fact]
+        public async Task PromiseResolve()
+        {
+            JObject input = new JObject
             {
-                return t.IsCompleted;
-            }, timeout: 5000, pollingInterval: 1000);
+                { "scenario", "promiseResolve" }
+            };
+
+            Task t = Fixture.Host.CallAsync("Scenarios",
+                new Dictionary<string, object>()
+                {
+                    { "input", input.ToString() }
+                });
 
-            // Await the task to force any exception to be thrown
-            await t;
+            await Task.WhenAny(t, Task.Delay(5000));
+            if (!t.IsCompleted) throw t.Exception;
```

Comment:
Argh... Had copied the statement from the code and just changed the property name. That's why I said *something like*. Well, you got the point :)